File: src/java/magma/Assignment.java
package magma;

import java.util.function.Function;

public record Assignment(Definable definition, String value) {
    Assignment mapDefinition(final Function<Definable, Definable> mapper) {
        final var definition1 = this.definition;
        final var definable2 = mapper.apply(definition1);
        return new Assignment(definable2, this.value());
    }

    String generate() {
        return this.definition().generate() + " = " + this.value();
    }
}

File: src/node/magma/Assignment.ts
/*import java.util.function.Function;*/
/*public record Assignment(Definable definition, String value) */ {
	Assignment constructor(/*final Function<Definable*//* Definable> mapper*/) {
		const definition1 = this.definition;
		const definable2 = mapper.apply(/*definition1*/);
		/*return new Assignment(definable2, this.value())*/;
	}
	String constructor() {
		let " : /*+*/ = /*" + this*/.value();
	}
	/**/}/**/
